plugins/UserId/Archiver.php

```diff
@@ -118,6 +118,21 @@ protected function insertDayReports()
     {
         /** @var DataTable $dataTable */
         $dataTable = $this->arrays->asDataTable();
+
+        // deal w/ ranking query summary row
+        $rankingQuerySummaryRow = $dataTable->getRowFromLabel(DataTable::LABEL_SUMMARY_ROW);
+        if ($rankingQuerySummaryRow) {
+            $rankingQuerySummaryRowId = $dataTable->getRowIdFromLabel(DataTable::LABEL_SUMMARY_ROW);
```

Comment:
can you explain why this code is needed, and do we do this kind of logic in other archivers too (which ones)? if not, why is it needed in this archiver?

Reply:
Ranking query creates a row w/ label -1, this is added to the DataArray which is just an array, not a datatable so it has no concept of a summary row. When DataArray::asDataTable() is called, it adds the row as a regular row and not as a summary row.

Reply:
Ok, wondering why is this logic not needed for other plugins?

Reply:
It is required, Actions does it here: https://github.com/matomo-org/matomo/blob/3.x-dev/plugins/Actions/ArchivingHelper.php#L383

It's not used in Events/Contents, but that's likely a bug that wasn't dealt with since it's not tested in BlobReportLimitingTest. I'll add it to my list to fix that.

Reply:
@diosmosis do you mind creating an issue for this bug in case it later slips off the list? unless you will open the PR this week or so. Thanks!